visualizations/frontend/filters/v2/TTagsInput.vue

```diff
@@ -0,0 +1,154 @@
+			value-key="value"
+		/>
+		<base-tags-input
+			v-else
+			v-model="selectedTags"
+			selection-only
+			:data="tags.length ? tags : tempTags"
+			:loading="loading"
+			:placeholder="placeholder"
+			@opened="handleOpen"
+			@update="handleUpdate"
+		/>
+	</div>
+</template>
+
+<script>
+	export default {
+		props: {
+			label: {
+				type: String,
+				default: '',
+			},
+			tKeyColumn: {
+				type: String,
+				default: 'KEY',
+			},
+			tValueColumn: {
+				type: String,
+				default: 'VALUE',
+			},
+			canReloadOnOpen: {
+				type: Boolean,
+				default: false,
+			},
+			isSingleTag: {
+				type: Boolean,
+				default: false,
+			},
+			placeholder: {
+				type: String,
+				default: '',
+			}
+		},
+		data: () => ({
+			is_filter: true,
+			selectedTags: [],
+			initialized: false,
+			tempTags: [], // base-multiple-value-input does not show selected items if `tags` is empty because data is fetched on open.
+		}),
+		computed:{ 
+			tags() {
+				const items = [];
+
+				if (this.rows.length) {
+					for (let row of this.rows) {
+						const key = row[this.tKeyColumn];
+						const value = row[this.tValueColumn];
+						if (key && value) {
+							items.push({
+								key: key.value,
+								value: value.value,
+							})
+						}
+					}
+				}
+				return items;
+			},
+			firstKey() {
+				// Used for single tag
+				if (this.tags.length) {
+					return this.tags[0].key;
+				} else if (this.tempTags) {
+					return this.tempTags[0].key;
+				}
+				return null;
+			}
+		},
+		keyColumn() {
+			return this.tKeyColumn ? this.tKeyColumn : this.findColumnByTag('keys');
+		},
+		valueColumn() {
+			return this.tValueColumn ? this.tValueColumn : this.findColumnByTag('values');
+		},
+		methods: {
+			onVisualizationInit() {
+				const initial_value = this.getFilterValue("selected_items");
+				const selectedTags = [];
+				const tempTags = [];
+
+				if (initial_value) {
+					const urlTags = initial_value.split('|');
+					for (let item of urlTags) {
+						const splitPair = item.split('_');
+						const key = splitPair.length > 1 ? splitPair[1] : splitPair[0];
+						const value = splitPair[0];
+
+						if (key && value) {
+							if (this.isSingleTag) {
+								selectedTags.push(value);
+								tempTags.push({ key, value })
+							} else {
+								selectedTags.push({ key, value });
+							}
+						}
+					}
+				}
+				this.tempTags = tempTags;
+				this.selectedTags = selectedTags;
+			},
+			handleOpen() {
+				if (!this.initialized || this.canReloadOnOpen) {
+					this.fetchLayerData();
+					this.initialized = true;
+				}
+			},
+			handleUpdate(items) {
+				let urlValue = '';
+				for (let item of items) {
+					if (this.isSingleTag) {
+						if (!this.firstKey) break;
+						let urlString = `${item}_${this.firstKey}|`
+						urlValue += urlString;
+					} else {
+						let urlString = `${item.value}_${item.key}|`
+						urlValue += urlString;
+					}
+				}
+				this.setFilterValue("selected_items", urlValue, true);
```

Comment:
The third parameter has been removed on the async filters PR. It was used to `notify` other components about the filter set but we do not need that flag anymore since we are relying on Vuex.
```suggestion
				this.setFilterValue("selected_items", urlValue);
```